src/main/java/core/basesyntax/strategy/BalanceOperationHandler.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+
+public class BalanceOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int quantity = fruitTransaction.getQuantity();
+        Storage.fruitStorage.put(fruit, quantity);
+    }
+}
```

Comment:
The `fruitStorage` should be encapsulated. Direct access to the storage map should be avoided. Instead, provide methods in the `Storage` class to interact with the storage.